src/sql/workbench/contrib/notebook/browser/notebookExplorer/notebookSearch.ts

```diff
@@ -0,0 +1,693 @@
+					content: nls.localize('searchMaxResultsWarning', "The result set only contains a subset of all matches. Please be more specific in your search to narrow down the results."),
+					type: MessageType.WARNING
+				});
+			}
+
+			if (!hasResults) {
+				const hasExcludes = !!excludePatternText;
+				const hasIncludes = !!includePatternText;
+				let message: string;
+
+				if (!completed) {
+					message = nls.localize('searchInProgress', "Search in progress... - ");
+				} else if (hasIncludes && hasExcludes) {
+					message = nls.localize('noResultsIncludesExcludes', "No results found in '{0}' excluding '{1}' - ", includePatternText, excludePatternText);
+				} else if (hasIncludes) {
+					message = nls.localize('noResultsIncludes', "No results found in '{0}' - ", includePatternText);
+				} else if (hasExcludes) {
+					message = nls.localize('noResultsExcludes', "No results found excluding '{0}' - ", excludePatternText);
+				} else {
+					message = nls.localize('noResultsFound', "No results found. Review your settings for configured exclusions and check your gitignore files - ");
+				}
+
+				// Indicate as status to ARIA
+				aria.status(message);
+
+				const messageEl = this.clearMessage();
+				const p = dom.append(messageEl, $('p', undefined, message));
+
+				if (!completed) {
+					const searchAgainLink = dom.append(p, $('a.pointer.prominent', undefined, nls.localize('rerunSearch.message', "Search again")));
+					this.messageDisposables.push(dom.addDisposableListener(searchAgainLink, dom.EventType.CLICK, (e: MouseEvent) => {
+						dom.EventHelper.stop(e, false);
+						this.triggerSearchQueryChange(query, excludePatternText, includePatternText, triggeredOnType, searchWidget);
+					}));
+					// cancel search
+					dom.append(p, $('span', undefined, ' / '));
+					const cancelSearchLink = dom.append(p, $('a.pointer.prominent', undefined, nls.localize('cancelSearch.message', "Cancel Search")));
+					this.messageDisposables.push(dom.addDisposableListener(cancelSearchLink, dom.EventType.CLICK, (e: MouseEvent) => {
+						dom.EventHelper.stop(e, false);
+						this.cancelSearch();
+					}));
+				} else if (hasIncludes || hasExcludes) {
+					const searchAgainLink = dom.append(p, $('a.pointer.prominent', { tabindex: 0 }, nls.localize('rerunSearchInAll.message', "Search again in all files")));
+					this.messageDisposables.push(dom.addDisposableListener(searchAgainLink, dom.EventType.CLICK, (e: MouseEvent) => {
+						dom.EventHelper.stop(e, false);
+					}));
+				} else {
+					const openSettingsLink = dom.append(p, $('a.pointer.prominent', { tabindex: 0 }, nls.localize('openSettings.message', "Open Settings")));
+					this.addClickEvents(openSettingsLink, this.onOpenSettings);
+				}
+
+				if (completed) {
+					dom.append(p, $('span', undefined, ' - '));
+
+					const learnMoreLink = dom.append(p, $('a.pointer.prominent', { tabindex: 0 }, nls.localize('openSettings.learnMore', "Learn More")));
```

Comment:
Let's not add this link - it's just going to confuse users since it refers to the Vs code search